File: public/docs-assets/examples/languages/quran-list-translation.py
# Python Example: Translation List
# This example shows how to fetch available translations using the GlobalQuran API

import requests
import json

# API Endpoint
endpoint = 'https://api.globalquran.com/v1/list/translation?key=YOUR_API_KEY'

def fetch_translations():
    """
    Function to fetch available translations
    
    Returns:
        dict: API response data or None if error
    """
    try:
        response = requests.get(endpoint)
        response.raise_for_status()  # Raises an HTTPError for bad responses
        
        data = response.json()
        return data
        
    except requests.exceptions.RequestException as e:
        print(f"Error fetching translations: {e}")
        return None
    except json.JSONDecodeError as e:
        print(f"Error parsing JSON: {e}")
        return None

def display_translations(data):
    """
    Display translations in a formatted way
    
    Args:
        data (dict): API response data
    """
    if not data or 'translation' not in data:
        print("No translation data available")
        return
    
    translations = data['translation']
    print(f"Found {len(translations)} translations:")
    print("-" * 50)
    
    for translation_id, translation_info in translations.items():
        print(f"ID: {translation_id}")
        print(f"Name: {translation_info.get('name', 'N/A')}")
        print(f"Language: {translation_info.get('language', 'N/A')}")
        print(f"Author: {translation_info.get('author', 'N/A')}")
        print("-" * 30)

# Main execution
if __name__ == "__main__":
    print("Fetching available translations...")
    data = fetch_translations()
    
    if data:
        display_translations(data)
        
        # Example of using a specific translation
        if 'translation' in data:
            first_translation = list(data['translation'].keys())[0]
            print(f"\nExample usage with translation ID '{first_translation}':")
            print(f"https://api.globalquran.com/v1/surah/1/{first_translation}?key=YOUR_API_KEY")
    else:
        print("Failed to fetch translations")

# Expected Response Structure:
# {
#   "translation": {
#     "en.sahih": {
#       "name": "Sahih International",
#       "language": "English",
#       "author": "Sahih International"
#     },
#     "en.pickthall": {
#       "name": "Pickthall",
#       "language": "English", 
#       "author": "Mohammed Marmaduke William Pickthall"
#     }
#   }
# }

File: public/docs-assets/examples/languages/quran-list-translation.js
// JavaScript Example: Translation List
// This example shows how to fetch available translations using the GlobalQuran API

// API Endpoint
const endpoint = 'https://api.globalquran.com/v1/list/translation?key=REPLACE_WITH_YOUR_KEY';

// Function to fetch translations
async function fetchTranslations() {
    try {
        const response = await fetch(endpoint);
        
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        
        const data = await response.json();
        return data;
    } catch (error) {
        console.error('Error fetching translations:', error);
        return null;
    }
}

// Function to display translations
function displayTranslations(data) {
    if (!data || !data.translation) {
        console.log('No translation data available');
        return;
    }
    
    const translations = data.translation;
    console.log('Available Translations:');
    console.log('=======================');
    
    Object.keys(translations).forEach(translationId => {
        const translation = translations[translationId];
        console.log(`ID: ${translationId}`);
        console.log(`Name: ${translation.name || 'N/A'}`);
        console.log(`Language: ${translation.language || 'N/A'}`);
        console.log(`Author: ${translation.author || 'N/A'}`);
        console.log('---');
    });
}

// Function to get translation by ID
function getTranslationById(data, translationId) {
    if (!data || !data.translation || !data.translation[translationId]) {
        return null;
    }
    
    return data.translation[translationId];
}

// Function to filter translations by language
function getTranslationsByLanguage(data, language) {
    if (!data || !data.translation) {
        return [];
    }
    
    return Object.entries(data.translation)
        .filter(([id, translation]) => translation.language === language)
        .map(([id, translation]) => ({ id, ...translation }));
}

// Function to get translations by author
function getTranslationsByAuthor(data, author) {
    if (!data || !data.translation) {
        return [];
    }
    
    return Object.entries(data.translation)
        .filter(([id, translation]) => translation.author && translation.author.includes(author))
        .map(([id, translation]) => ({ id, ...translation }));
}

// Function to search translations by name
function searchTranslations(data, searchTerm) {
    if (!data || !data.translation) {
        return [];
    }
    
    const term = searchTerm.toLowerCase();
    return Object.entries(data.translation)
        .filter(([id, translation]) => 
            translation.name && translation.name.toLowerCase().includes(term)
        )
        .map(([id, translation]) => ({ id, ...translation }));
}

// Main execution
async function main() {
    console.log('Fetching available translations...');
    const data = await fetchTranslations();
    
    if (data) {
        displayTranslations(data);
        
        // Example: Get specific translation
        const translation = getTranslationById(data, 'en.sahih');
        if (translation) {
            console.log('\nSpecific Translation (en.sahih):');
            console.log(`Name: ${translation.name}`);
            console.log(`Language: ${translation.language}`);
            console.log(`Author: ${translation.author}`);
        }
        
        // Example: Filter by language
        const englishTranslations = getTranslationsByLanguage(data, 'English');
        console.log(`\nEnglish Translations: ${englishTranslations.length}`);
        
        // Example: Search by name
        const sahihTranslations = searchTranslations(data, 'sahih');
        console.log(`Translations containing "sahih": ${sahihTranslations.length}`);
        
        console.log('\nExample usage:');
        console.log('const translations = await fetchTranslations();');
        console.log('const translation = getTranslationById(translations, "en.sahih");');
        console.log('const englishTranslations = getTranslationsByLanguage(translations, "English");');
        console.log('const searchResults = searchTranslations(translations, "sahih");');
    } else {
        console.log('Failed to fetch translations');
    }
}

// Run the example
main();

// Expected Response Structure:
// {
//   "translation": {
//     "en.sahih": {
//       "name": "Sahih International",
//       "language": "English",
//       "author": "Sahih International"
//     },
//     "en.pickthall": {
//       "name": "Pickthall",
//       "language": "English", 
//       "author": "Mohammed Marmaduke William Pickthall"
//     },
//     "ar.jalalayn": {
//       "name": "Tafsir al-Jalalayn",
//       "language": "Arabic",
//       "author": "Jalal ad-Din al-Mahalli and Jalal ad-Din as-Suyuti"
//     }
//   }
// }
